DBSCE/Overview.md

```diff
@@ -0,0 +1,221 @@
+
+
+### Local Key Helper
+
+DBSC(E) introduces the concept of `Local Key Helper`.
+
+**Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. *Local key helper* can be public or private and is expected to be either shipped as a part of a given enterprise framework (with the IdP/OS) or can be installed by a provider in compliance with the protocol expanded below.
+
+From the deployment point of view there are two types of local key helpers: _well-known_(_private_) and _third party_(_public_)
+
+- _Public local key helper_ or _third party_: Can be accessed by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP through as defined in DBSC(E) protocol. A third party local key helper has a special deployment mechanism. We expect to loosely define the API and deployment mechanism in the upcoming specification, and will leave the specific implementation details to the vendors. However, we have provided examples of how this can look for a few platforms [below](#platform-examples).
+- _Private local key helper_ or _well known_ : Can be specific to an IdP. Typically owned by the IdP and will have a private protocol to communicate with the IdP. Comes with either OS or built into the browser. It assumed that well-known key helpers are **trusted and enabled by default** in a browser and/or a given IDP. A browser knows how to activate a well-known key helper.
+
+The Local Key Helper is responsible for:
+
+- Generation of the binding key and producing binding statements (see below)
+- Producing signatures with the binding key
+- Cleanup of the binding key and its artifacts (when the user clears the browser session or the key is unused for a long time).
+
+#### Platform Examples
+
+- Windows: Please refer to the Windows Local Key Helper [here](./KeyGeneration.md#local-key-helper-on-windows) for an example of a Local Key Helper.
+- MacOS: TBD
+- Android:TBD
+
+Note: Above are examples of Local Key Helpers that can be used for DBSC(E) key generation. Any platform is free to build a version of this, as long as the API is consistent with the protocol.
+
+### Attestation Service:
+
+A service that is responsible for verifying that the binding key is issued by the expected device and providing the attestation of the key to the IdP. The attestation service can be owned by the IdP or a third party. DBSC(E) relies on the attestation service to validate the binding statement and ensure that the binding key and the device key belong to the same device. We have added details on the specifics of the binding artifacts generated during the device registration process, and the validation of the binding statement in the [DBSC(E) Key Generation](#key-generation-specifics) section.
+
+### Device Registration (Pre-Session)
```

Comment:
I suggest moving this up right after "device registration client" for continuation. Also, I'd recommend removing pre-session int he title.

Reply:
Done. Ideally, since we establish this as a pre-requisite for any flow, we may need to mention it in the design that it is expected this step is completed. I will add the same.

Reply:
Reopening this. I have moved it up, but since AIK is not defined by the time device registration client is defined, I still think the right place is after Attestation Service and Key definitions are covered. Let us close this offline.

Reply:
Closing. Adding glossary and we can revisit once all sections are checked for uniformity.